### PR TITLE
ReadOnly streams shouldn't be flushed

### PR DESCRIPTION
**Pull request by osoykan**
I observed the behavior that [ProxyMiddleware flushes every stream](https://github.com/damianh/ProxyKit/blob/master/src/ProxyKit/ProxyMiddleware.cs#L56) without having a look if it is read-only or not. 

So, sometimes AspnetCore TestsHost creates read-only streams as the response for FOUND verbs I don't know exactly when but, after upgrading ProxyKit the latest version [this change](https://github.com/damianh/ProxyKit/commit/7c893970448ff519f9f520fa2d1ce7f0287ff9c8#diff-bf059ff7bcbac62b87c51b955d0e8e7c) led to the problem and I got `The stream does not support writing.
   at System.Net.Http.StreamContent.ReadOnlyStream.FlushAsync(CancellationToken cancellationToken)` exception, since it is read-only.

This change will address the issue.

### CODE REVIEW

File: src/ProxyKit.Tests/PassthroughProxyTest.cs [hunk 1]
```diff
@@ -177,6 +178,43 @@ public async Task X_Forwarded_Headers_should_be_removed_by_default()
...
+        public async Task Response_stream_should_not_be_Flushed_if_the_response_is_ReadyOnly()
+        {
+            _testMessageHandler = new TestMessageHandler
+            {
+                Sender = req =>
+                {
+                    var response = new HttpResponseMessage(HttpStatusCode.Found)
+                    {
+                        // Usually the response of FOUND verb comes with null stream in TestHost. At least that's been observed sometimes.
+                        Content = new StreamContent(Stream.Null)
```
Comment (osoykan) [inline src/ProxyKit.Tests/PassthroughProxyTest.cs]: mimic the behavior I had.

**Comment by nmocruz** [comment]
I have a similar issue, for some reason is getting and EmptyReadStream and trying to flush
`System.NotSupportedException: Specified method is not supported.
   at System.Net.Http.EmptyReadStream.FlushAsync(CancellationToken cancellationToken)
   at ProxyKit.ProxyMiddleware`1.CopyProxyHttpResponse(HttpContext context, HttpResponseMessage responseMessage)
   at ProxyKit.ProxyMiddleware`1.Invoke(HttpContext context)`

**Comment by damianh** [comment]
Thanks for PR @osoykan . Merging and will do a patch release soon.

**Comment by osoykan** [comment]
Thanks for the review @damianh. looking forward to it.

**Pull request merged**